src/Pact/Repl.hs

```diff
@@ -322,15 +322,25 @@ execScript' m fp = do
   runStateT (useReplLib >> loadFile fp) s
 
 
+-- | install repl lib functions into monad state
 useReplLib :: Repl ()
 useReplLib = id %= setReplLib
 
+-- | mutate repl state to install lib functions
 setReplLib :: ReplState -> ReplState
-setReplLib = rEvalState.evalRefs.rsLoaded %~ HM.union (moduleToMap replDefs)
+setReplLib = over (rEvalState.evalRefs.rsLoaded) $ HM.union (moduleToMap replDefs)
 
+-- | mutate repl state to remove lib functions
+unsetReplLib :: ReplState -> ReplState
+unsetReplLib = over (rEvalState.evalRefs.rsLoaded) (`HM.difference` (moduleToMap replDefs))
 
+-- | evaluate string in repl monad
+evalPact :: String -> Repl (Either String (Term Name))
+evalPact cmd = parsedCompileEval cmd (TF.parseString exprsOnly mempty cmd)
```

Comment:
If the point of this function is to guarantee that `cmd` does not have access to REPL functions, do we need to call unsetReplLib in this function to ensure that it is unset?

Reply:
It's used in `evalRepl'` below so I don't think that will work.

Reply:
Oh right, good point.  My initial mental model was that this is a higher level function to ensure that the repl lib is NOT available.  But because `evalRepl'` calls `useReplLib` first, that implies that this function is the lower level generic function.  That makes sense.  Maybe we should state something to this effect in the comments?

Reply:
No, it is simply to provide raw access to the parse-compile-eval machinery and not re-set the repl lib every time like `evalRepl'`, which is inefficient. The most sophisticated approach is to directly control when the repl lib is available and only use `evalPact`, but the more incremental change is to only use `evalPact` after removing lib with `unsetReplLib`, and otherwise use `evalRepl'`.

Reply:
Once web-repl is stable and we've rolled out the next Pact version, we should do a review of the repl functions, as it appears `evalString` is not needed anymore (which was built for the original `try-pact`), get explicit exports, etc.

Reply:
This would be the time to further rationalize how locked-down ReplState interactions should be, etc.